lib/internal/http2/compat.js

```diff
@@ -181,6 +181,10 @@ class Http2ServerRequest extends Readable {
     return '2.0';
   }
 
+  get socket() {
+    return this.stream.session.socket;
+  }
```

Comment:
> If we explicitly do not implement this part of the compatibility layer, maybe it should throw or emit a warning?

I think we should just throw. The meaning is so different that we can't really provide an equivalent. If you use the socket directly with `require('http')`, very likely your code will not work with `require('http2')`.

> In this case I'm using it to report the alpnProtocol from the underlying TLS connection. I think there could be other valid use cases. $0.02

You need this just for tests?

https://github.com/nodejs/http2/pull/125/files#diff-23822d26e014a01b71d6319c40bef2a6R27

I do not think that is a valid use case.